store/src/main/java/org/apache/rocketmq/store/queue/CombineConsumeQueueStore.java

```diff
@@ -0,0 +1,527 @@
+            store.recover(concurrently);
+        }
+        log.info("CombineConsumeQueueStore recover success, concurrently={}", concurrently);
+    }
+
+    @Override
+    public boolean isMappedFileMatchedRecover(long phyOffset, long storeTimestamp,
+        boolean recoverNormally) throws RocksDBException {
+        if (!assignOffsetStore.isMappedFileMatchedRecover(phyOffset, storeTimestamp, recoverNormally)) {
+            return false;
+        }
+
+        for (AbstractConsumeQueueStore store : innerConsumeQueueStoreList) {
+            if (store == assignOffsetStore || store.isMappedFileMatchedRecover(phyOffset, storeTimestamp, recoverNormally)) {
+                continue;
+            }
+
+            if (toleranceLookBackFailuresNum.getAndIncrement() <= 0) {
+                // toleranceLookBackFailuresNum <= 0, so only can read from assignOffsetStore
+                if (assignOffsetStore != currentReadStore) {
+                    log.error("CombineConsumeQueueStore currentReadStore not satisfied readable conditions, assignOffsetStore={}, currentReadStore={}",
+                        assignOffsetStore.getClass().getSimpleName(), currentReadStore.getClass().getSimpleName());
+                    throw new IllegalArgumentException(store.getClass().getSimpleName() + " not satisfied readable conditions, only can read from " + assignOffsetStore.getClass().getSimpleName());
+                }
+                log.warn("CombineConsumeQueueStore can not recover all inner store, maybe some inner store start haven’t started before, store={}",
+                    store.getClass().getSimpleName());
+                return true;
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public long getDispatchFromPhyOffset() {
+        long dispatchFromPhyOffset = assignOffsetStore.getDispatchFromPhyOffset();
+        for (AbstractConsumeQueueStore store : innerConsumeQueueStoreList) {
+            if (store == assignOffsetStore) {
+                continue;
+            }
+            if (store.getDispatchFromPhyOffset() < dispatchFromPhyOffset) {
+                dispatchFromPhyOffset = store.getDispatchFromPhyOffset();
+            }
+        }
+        return dispatchFromPhyOffset;
+    }
+
+    @Override
+    public void start() {
+        boolean success = false;
+        try {
+            success = checkAssignOffset(true);
+        } catch (RocksDBException e) {
+            log.error("CombineConsumeQueueStore checkAssignOffsetStore fail", e);
+        }
+
+        if (!success && assignOffsetStore != currentReadStore) {
+            throw new RuntimeException("CombineConsumeQueueStore checkAssignOffsetStore fail");
+        }
+
+        for (AbstractConsumeQueueStore store : innerConsumeQueueStoreList) {
+            store.start();
+        }
+    }
+
+    public boolean checkAssignOffset(boolean initializeOffset) throws RocksDBException {
```

Comment:
how about rename to initAssginOffsetForAllConsumeQueueStores， and add javadoc here for readability.